base/buffer_vector.hpp

```diff
@@ -71,7 +74,7 @@ template <class T, size_t N> class buffer_vector
 
   buffer_vector(buffer_vector const &) = default;
 
-  buffer_vector(buffer_vector && rhs) : m_size(rhs.m_size), m_dynamic(std::move(rhs.m_dynamic))
+  buffer_vector(buffer_vector && rhs) noexcept : m_size(rhs.m_size), m_dynamic(std::move(rhs.m_dynamic))
```

Comment:
Well, yes, actually it should depend on ```T(T&&) noexcept```

1. Simple one: add ```static_assert(noexcept(T()));```
2. Complex one: add ```enable_if blah-blah```

Reply:
`noexcept(std::is_nothrow_constructible_v<T, T>)`?

Reply:
Looks good. Does it check T() and T(T&&) ?

Reply:
I'm not sure about `noexcept(std::is_nothrow_constructible_v<T, T>)`
I'd better revert this change, because I don't have time to investigate it right now

Reply:
Could you please try the simple:
```
static_assert(std::is_nothrow_move_constructible<T>::value);
```

Reply:
I did and pushed it myself.